extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/DevConsoleCORSFilter.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.cors.CORSConfig;
+import io.quarkus.vertx.http.runtime.cors.CORSFilter;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class DevConsoleCORSFilter implements Handler<RoutingContext> {
+    private static final Logger LOG = Logger.getLogger(DevConsoleCORSFilter.class);
+
+    private static final String HTTP_PORT_CONFIG_PROP = "quarkus.http.port";
+    private static final String LOCAL_HOST = "localhost";
+    private static final String HTTP_LOCAL_HOST = "http://" + LOCAL_HOST;
+    private static final String HTTPS_LOCAL_HOST = "https://" + LOCAL_HOST;
+
+    public DevConsoleCORSFilter() {
+    }
+
+    private static CORSFilter corsFilter() {
+        int httpPort = ConfigProvider.getConfig().getValue(HTTP_PORT_CONFIG_PROP, int.class);
+        CORSConfig config = new CORSConfig();
+        config.origins = Optional.of(List.of(HTTP_LOCAL_HOST + ":" + httpPort, HTTPS_LOCAL_HOST + ":" + httpPort));
+        return new CORSFilter(config);
+    }
+
+    @Override
+    public void handle(RoutingContext event) {
+        HttpServerRequest request = event.request();
+        HttpServerResponse response = event.response();
+        String origin = request.getHeader(HttpHeaders.ORIGIN);
+        if (origin == null) {
+            corsFilter().handle(event);
+        } else {
+            if (origin.startsWith(HTTP_LOCAL_HOST) || origin.startsWith(HTTPS_LOCAL_HOST)) {
+                corsFilter().handle(event);
+            } else {
+                LOG.error("Only localhost origin is allowed");
+                response.end();
```

Comment:
I think we need to produce an error, and not returning an empty 200 response.

Reply:
@cescoffier Here we work against the browser security which will report an error because Quarkus will not return an expected allowed origin header

Reply:
Well, it is really a job of the browser security manager to report the error, it is not the status but the allowed origins that will let the browser handle the request. One other thing about `200` is that sometimes `200` is returned so as to not to give any hints at all in case of security issues. Not sure if this is what the original intention was.

`403` won't make it worse but it'll have to be done in the main CORS filter as well, so it seems it is a follow up kind of PR if `403` is the status that will have to be returned.

Reply:
I'll get it updated to return 403

Reply:
@cescoffier, @geonard, both this new handler and the main CORS handler will now return `403` if the origins check has failed, CC @stuartwdouglas

Reply:
I thought I'd avoid setting the actual text message for now, `403` is here now, and the browser must be having its own cache of messages depending on the availability of `Allow-*` headers. If you'd like we can add `CORS Rejected - Invalid origin` - but it does feel it is not really necessary. That said - it can be done too if you'd like :-)

Reply:
Oh well, just ignore me, if it is a simple request with the wrong origin then `403` alone would not help. Let me add this message, sorry for overloading CI with my pushes